main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -574,13 +575,62 @@ object ConstraintGen {
         (resTpe, resEff)
 
       case Expr.StructNew(sym, fields, region, tvar, evar, loc) =>
-        throw new RuntimeException("joe tbd")
+        val (instantiatedFieldTpes, structTpe, regionVar) = instantiateStruct(sym, root.structs)
+        val visitedFields = fields.map { case (k, v) => visitExp(v) }
+        val (fieldTpes, fieldEffs) = visitedFields.unzip
+        // struct type
+        c.unifyType(tvar, structTpe, loc)
+        // field types
+        fields.zip(fieldTpes).foreach {
+          case ((fieldSym, expr), fieldTpe1) =>
+            instantiatedFieldTpes.get(fieldSym) match {
+              case None => () // Field does not exist. A resolution error will already have been generated
+              case Some(fieldTpe2) => c.unifyType(fieldTpe1, fieldTpe2, expr.loc)
+            }
+        }
+        // region type
+        val regionType = Type.mkRegion(regionVar, loc)
+        val (regionTpe, regionEff) = visitExp(region)
+        c.unifyType(regionType, regionTpe, region.loc)
+        // effect
```

Comment:
Drop this comment (and similar comments like "overall effect".

Reply:
I think, overall, comments should either be more descriptive (longer) or not there at all.